tx-indexer/src/storage/database.rs

```diff
@@ -188,15 +188,17 @@ impl HashStorageWithDB {
 #[async_trait::async_trait]
 impl TxCollectingStorage for HashStorageWithDB {
     async fn restore_transaction_by_receipt_id(&self, receipt_id: &str) -> anyhow::Result<()> {
-        let transaction_details = self
+        if let Ok(transaction_details) = self
```

Comment:
By this change, you've stopped returning the error from `get_transaction_by_receipt_id()`, is this intentional? Why?

Reply:
We should not try to restore a transaction that is missing from the storage cache. This became possible because we have a mechanism for indexing transactions only linked to a specific accounts.